src/Symfony/Component/PropertyAccess/Exception/UnexpectedTypeException.php

```diff
@@ -18,8 +18,4 @@
  */
 class UnexpectedTypeException extends RuntimeException
 {
-    public function __construct($value, $expectedType)
```

Comment:
I'd suggest to
1. change the exception in PropertyPath to `Symfony\Component\PropertyAccess\Exception\InvalidArgumentException` because that is more consistent with https://github.com/mpajunen/symfony/blob/property-access-exception-messages/src/Symfony/Component/PropertyAccess/PropertyAccessor.php#L57 and also makes more sense since it is a RuntimeException isntead of a LogicException (and should not be considered BC break since it's not intended to be catched).
2. adjust UnexpectedTypeException  with optional extra params